packages/plugins/i18n/server/src/bootstrap.ts

```diff
@@ -14,6 +14,30 @@ const registerModelsHooks = () => {
       await getService('permissions').actions.syncSuperAdminPermissionsWithLocales();
     },
   });
+
+  const i18nModelUIDs = Object.values(strapi.contentTypes)
+    .filter((contentType) => getService('content-types').isLocalizedContentType(contentType))
+    .map((contentType) => contentType.uid);
+
+  strapi.documents.use(async (context, next) => {
+    // @ts-expect-error ContentType is not typed correctly on the context
+    const schema = context.contentType;
+
+    if (!['create', 'update'].includes(context.action)) {
+      return next(context);
+    }
+
+    if (!i18nModelUIDs.includes(schema.uid)) {
+      return next(context);
+    }
```

Comment:
We could use the isLocalizedContentType directly here, wdyt?
```suggestion
    if (!igetService('content-types').isLocalizedContentType(schema)) {
      return next(context);
    }
```